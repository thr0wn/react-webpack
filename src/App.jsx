import React from "react";
import { Redirect, Route, Switch } from "react-router-dom";
import Home from "@/src/pages/Home";

const App = () => (
  <Switch>
    <Route path="/home">
      <Home />
    </Route>
    <Redirect from="/*" to="/home" exact />
  </Switch>
);

export default App;
