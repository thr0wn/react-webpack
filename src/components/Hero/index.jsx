import React from "react";
import PropTypes from "prop-types";
import { Button } from "@material-ui/core";
import StyledHero, {
  StyledText,
  StyledLeftContent,
  StyledRightContent,
  StyledImg,
} from "./index.styles";

const Hero = ({ marked }) => (
  <StyledHero marked={marked}>
    <StyledLeftContent>
      <div>
        <StyledText>Lorem ipsum dolor piscing</StyledText>
        <StyledText>sit amet, consectetur</StyledText>
        <StyledText marked>adipiscing elit</StyledText>
        <Button>Teste gr√°tis por 30 dias</Button>
      </div>
    </StyledLeftContent>
    <StyledRightContent>
      <StyledImg />
    </StyledRightContent>
  </StyledHero>
);

Hero.propTypes = {
  marked: PropTypes.bool,
};

Hero.defaultProps = {
  marked: false,
};

export default Hero;
