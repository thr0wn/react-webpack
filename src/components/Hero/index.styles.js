import styled, { css } from "styled-components";
import heroImg from "./hero.png";

export const StyledLeftContent = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: flex-end;
  padding: 48px 48px 72px 109px;
  width: 44%;

  @media (max-width: 800px) {
    width: 100%;
    align-items: center;
    justify-content: center;
    padding: 16px;
  }

  button {
    margin-top: 24px;
  }
`;

export const StyledImg = styled.div`
  height: 100%;
  background-image: url(${heroImg});
  background-position: center;
  background-repeat: no-repeat;
  background-size: cover;
`;

export const StyledRightContent = styled.div`
  width: 56%;

  @media (max-width: 800px) {
    width: 0%;
  }
`;

export const StyledText = styled.p`
  font-size: 48px;
  font-style: normal;
  font-weight: 300;
  line-height: 77px;
  letter-spacing: 0px;
  text-align: left;
  width: max-content;
  position: relative;
  z-index: 1;

  @media (max-width: 1373px) {
    font-size: 40px;
  }

  @media (max-width: 1204px) {
    font-size: 28px;
  }

  @media (max-width: 1078px) {
    font-size: 32px;
  }

  ${({ marked }) =>
    marked &&
    css`
      white-space: nowrap;

      &::before {
        content: "";
        height: 100%;
        background: #fde7e7;
        position: absolute;
        top: 0;
        bottom: 0;
        left: -100vw;
        right: -20px;
        z-index: -1;

        @media (max-width: 800px) {
          right: -20px;
        }
      }
    `}
`;

export default styled.div`
  display: flex;
  position: relative;
  height: 515px;
  max-width: 100vw;

  @media (max-width: 800px) {
    height: auto;
  }
`;
