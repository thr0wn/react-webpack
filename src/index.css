body {
  margin: 0;
  padding: 0;
}

.text {
  font-size: 16px;
  font-style: normal;
  font-weight: 500;
  line-height: 26px;
  letter-spacing: 0px;
  text-align: right;
  color: black;
}

.link {}
