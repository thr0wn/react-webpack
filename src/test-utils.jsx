import React from "react";
import { render } from "@testing-library/react";
import renderer from "react-test-renderer";
import axios from "axios";
import MockAdapter from "axios-mock-adapter";

const AllTheProviders = ({ children }) => {
  return <>{children}</>;
};

const customRender = (ui, options) =>
  render(ui, { wrapper: AllTheProviders, ...options });

const renderAsJSON = (children) =>
  renderer.create(<AllTheProviders>{children}</AllTheProviders>).toJSON();

export * from "@testing-library/react";

export { customRender as render, renderAsJSON };

jest.mock("react-redux", () => ({
  ...jest.requireActual("react-redux"),
  useSelector: jest.fn(),
}));

export const axiosMock = new MockAdapter(axios);
