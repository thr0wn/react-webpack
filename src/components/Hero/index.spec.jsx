import React from "react";
import { render } from "@/src/test-utils";
import Hero from ".";

describe("Hero", () => {
  let rendered;

  beforeEach(() => {
    rendered = render(<Hero />);
  });

  it("renders correctly", () => {
    const { container } = rendered;
    expect(container).toMatchSnapshot();
  });

  it("renders correctly with marked=true", () => {
    const { container } = render(<Hero marked />);
    expect(container).toMatchSnapshot();
  });

  it("renders cta button", () => {
    const button = rendered.getByText("Teste gr√°tis por 30 dias");
    expect(button).toBeVisible();
    expect(button).toBeEnabled();
  });
});
