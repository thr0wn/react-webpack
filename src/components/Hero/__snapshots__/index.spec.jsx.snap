// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hero renders correctly 1`] = `
.c4 {
  width: auto;
  min-width: 64px;
  height: 36px;
  padding: 6px 16px;
  box-shadow: none;
  font-size: 0.875rem;
  font-weight: 400;
  line-height: 1.5;
  background: #a82827;
  color: #ffffff;
  text-transform: none;
}

.c4:hover {
  background: #d95857;
}

.c4:active {
  background: #b72c2a;
  box-shadow: none;
}

.c4:focus {
  background: #b72c2a;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  padding: 48px 48px 72px 109px;
  width: 44%;
}

.c1 button {
  margin-top: 24px;
}

.c6 {
  height: 100%;
  background-image: url();
  background-position: center;
  background-repeat: no-repeat;
  background-size: cover;
}

.c5 {
  width: 56%;
}

.c2 {
  font-size: 48px;
  font-style: normal;
  font-weight: 300;
  line-height: 77px;
  -webkit-letter-spacing: 0px;
  -moz-letter-spacing: 0px;
  -ms-letter-spacing: 0px;
  letter-spacing: 0px;
  text-align: left;
  width: -webkit-max-content;
  width: -moz-max-content;
  width: max-content;
  position: relative;
  z-index: 1;
}

.c3 {
  font-size: 48px;
  font-style: normal;
  font-weight: 300;
  line-height: 77px;
  -webkit-letter-spacing: 0px;
  -moz-letter-spacing: 0px;
  -ms-letter-spacing: 0px;
  letter-spacing: 0px;
  text-align: left;
  width: -webkit-max-content;
  width: -moz-max-content;
  width: max-content;
  position: relative;
  z-index: 1;
  white-space: nowrap;
}

.c3::before {
  content: "";
  height: 100%;
  background: #fde7e7;
  position: absolute;
  top: 0;
  bottom: 0;
  left: -100vw;
  right: -20px;
  z-index: -1;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: relative;
  height: 515px;
  max-width: 100vw;
}

@media (max-width:800px) {
  .c1 {
    width: 100%;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
    padding: 16px;
  }
}

@media (max-width:800px) {
  .c5 {
    width: 0%;
  }
}

@media (max-width:1373px) {
  .c2 {
    font-size: 40px;
  }
}

@media (max-width:1204px) {
  .c2 {
    font-size: 28px;
  }
}

@media (max-width:1078px) {
  .c2 {
    font-size: 32px;
  }
}

@media (max-width:1373px) {
  .c3 {
    font-size: 40px;
  }
}

@media (max-width:1204px) {
  .c3 {
    font-size: 28px;
  }
}

@media (max-width:1078px) {
  .c3 {
    font-size: 32px;
  }
}

@media (max-width:800px) {
  .c3::before {
    right: -20px;
  }
}

@media (max-width:800px) {
  .c0 {
    height: auto;
  }
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <div>
        <p
          class="c2"
        >
          Lorem ipsum dolor piscing
        </p>
        <p
          class="c2"
        >
          sit amet, consectetur
        </p>
        <p
          class="c3"
        >
          adipiscing elit
        </p>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text c4"
          tabindex="0"
          type="button"
          variation="primary"
        >
          <span
            class="MuiButton-label"
          >
            Teste grátis por 30 dias
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
    <div
      class="c5"
    >
      <div
        class="c6"
      />
    </div>
  </div>
</div>
`;

exports[`Hero renders correctly with marked=true 1`] = `
.c4 {
  width: auto;
  min-width: 64px;
  height: 36px;
  padding: 6px 16px;
  box-shadow: none;
  font-size: 0.875rem;
  font-weight: 400;
  line-height: 1.5;
  background: #a82827;
  color: #ffffff;
  text-transform: none;
}

.c4:hover {
  background: #d95857;
}

.c4:active {
  background: #b72c2a;
  box-shadow: none;
}

.c4:focus {
  background: #b72c2a;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  padding: 48px 48px 72px 109px;
  width: 44%;
}

.c1 button {
  margin-top: 24px;
}

.c6 {
  height: 100%;
  background-image: url();
  background-position: center;
  background-repeat: no-repeat;
  background-size: cover;
}

.c5 {
  width: 56%;
}

.c2 {
  font-size: 48px;
  font-style: normal;
  font-weight: 300;
  line-height: 77px;
  -webkit-letter-spacing: 0px;
  -moz-letter-spacing: 0px;
  -ms-letter-spacing: 0px;
  letter-spacing: 0px;
  text-align: left;
  width: -webkit-max-content;
  width: -moz-max-content;
  width: max-content;
  position: relative;
  z-index: 1;
}

.c3 {
  font-size: 48px;
  font-style: normal;
  font-weight: 300;
  line-height: 77px;
  -webkit-letter-spacing: 0px;
  -moz-letter-spacing: 0px;
  -ms-letter-spacing: 0px;
  letter-spacing: 0px;
  text-align: left;
  width: -webkit-max-content;
  width: -moz-max-content;
  width: max-content;
  position: relative;
  z-index: 1;
  white-space: nowrap;
}

.c3::before {
  content: "";
  height: 100%;
  background: #fde7e7;
  position: absolute;
  top: 0;
  bottom: 0;
  left: -100vw;
  right: -20px;
  z-index: -1;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: relative;
  height: 515px;
  max-width: 100vw;
}

@media (max-width:800px) {
  .c1 {
    width: 100%;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
    padding: 16px;
  }
}

@media (max-width:800px) {
  .c5 {
    width: 0%;
  }
}

@media (max-width:1373px) {
  .c2 {
    font-size: 40px;
  }
}

@media (max-width:1204px) {
  .c2 {
    font-size: 28px;
  }
}

@media (max-width:1078px) {
  .c2 {
    font-size: 32px;
  }
}

@media (max-width:1373px) {
  .c3 {
    font-size: 40px;
  }
}

@media (max-width:1204px) {
  .c3 {
    font-size: 28px;
  }
}

@media (max-width:1078px) {
  .c3 {
    font-size: 32px;
  }
}

@media (max-width:800px) {
  .c3::before {
    right: -20px;
  }
}

@media (max-width:800px) {
  .c0 {
    height: auto;
  }
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <div>
        <p
          class="c2"
        >
          Lorem ipsum dolor piscing
        </p>
        <p
          class="c2"
        >
          sit amet, consectetur
        </p>
        <p
          class="c3"
        >
          adipiscing elit
        </p>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text c4"
          tabindex="0"
          type="button"
          variation="primary"
        >
          <span
            class="MuiButton-label"
          >
            Teste grátis por 30 dias
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
    <div
      class="c5"
    >
      <div
        class="c6"
      />
    </div>
  </div>
</div>
`;
